### PR TITLE
docs(examples): GDPR Cookie Consent

### PR DESCRIPTION
**Pull request by gavriguy**
This example demonstrates how to create a GDPR banner.
Once the user gives her consent, the app will start tracking the user data by enabling a script on the root

### CODE REVIEW

File: examples/gdpr-cookie-consent/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Create a simple GDPR consent form.
+Till the user doesn't click on the `Accept` button, she will see a banner prompting to accept cookies.
+
+
+## Preview
+
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+<!-- TODO: update this link to the path for your example: -->
+
```
Comment (MichaelDeBoey) [inline examples/gdpr-cookie-consent/README.md:12]:
```suggestion
```

File: examples/gdpr-cookie-consent/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/remix-gdpr-example-e7znt?file=/app/routes/index.tsx)
```
Comment (MichaelDeBoey) [inline examples/gdpr-cookie-consent/README.md:13]:
```suggestion
[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/gdpr-cookie-consent)
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: examples/gdpr-cookie-consent/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Preview
+
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/gdpr-cookie-consent)
+
+## Example
+
+Users will be presented with a GDPR consent form on every page ([app/routes/index.tsx](app/routes/index.tsx)) till they submit the accept button.
```
Comment (sergiodxa) [inline examples/gdpr-cookie-consent/README.md:15]: If the form is in the index and not the root, then it’s not going to be on every page

File: examples/gdpr-cookie-consent/app/root.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        <Form method="post" reloadDocument>
+         We use Cookies...
+         {/* You can pass values on the submission button  */}
+         <button name="accept-gdpr" value="true" type="submit">Accept</button>
+       </Form>
+    </div>}
+        <ScrollRestoration />
+        { track && <script src="/dummy-analytics-script.js"></script> }
+        <Scripts />
+        {process.env.NODE_ENV === "development" && <LiveReload />}
```
Comment (MichaelDeBoey) [inline examples/gdpr-cookie-consent/app/root.tsx:65]:
```suggestion
        <LiveReload />
```
See #1352

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: examples/gdpr-cookie-consent/app/root.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      </head>
+      <body>
+        <Outlet />
+        {!track && <div style={{
+      backgroundColor: '#ccc',
+      padding: 10,
+      position: 'fixed',
+      bottom: 0,
+     }}>
+        <Form method="post" reloadDocument>
```
Comment (kentcdodds) [inline examples/gdpr-cookie-consent/app/root.tsx:56]: Can we avoid the use of `reloadDocument`?

File: examples/gdpr-cookie-consent/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+## Preview
+
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/gdpr-cookie-consent)
+
+## Example
+
+Users will be presented with a GDPR consent form on every page [app/root.tsx](app/root.tsx) till they submit the accept button.
+Once they submit the consent form, a dummy tracking [script](public/dummy-analytics-script.js) at the [app/root.tsx](app/root.tsx)  will start tracking the user`s data (open the browser console too see the dummy tracking message).
```
Comment (kentcdodds) [inline examples/gdpr-cookie-consent/README.md:16]:
```suggestion
Once they submit the consent form, a dummy tracking [script](public/dummy-analytics-script.js) at the [app/root.tsx](app/root.tsx) will start tracking the user's data (open the browser console too see the dummy tracking message).
```

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Looks good to me. I wonder if we could avoid `reloadDocument`. Is there a reason that's there?

File: examples/gdpr-cookie-consent/app/root.tsx (hunk 5 above)
Comment (gavriguy) [reply examples/gdpr-cookie-consent/app/root.tsx:56]: Without it the code on public/dummy-analytics-script.js will not run immediately after the form submission.
Is there a better way of doing it?

Comment (kentcdodds) [reply examples/gdpr-cookie-consent/app/root.tsx]: Yeah, looks like inserting a script tag with React doesn't work, so we have to use `document.createElement` in a `useEffect` which is fine. I've made that update myself (hope you don't mind) and I also changed it to use `useFetcher` so we don't get a navigation and add to the history stack.

**Review by kentcdodds — APPROVED** [review]
Thanks for working on this!

File: examples/gdpr-cookie-consent/app/root.tsx (hunk 5 above)
Comment (gavriguy) [reply examples/gdpr-cookie-consent/app/root.tsx]: Thank you :)

**Pull request merged**